examples/plotting/file/eclipse.py

```diff
@@ -0,0 +1,75 @@
+# Based on https://www.reddit.com/r/dataisbeautiful/comments/6qnkg0/google_search_interest_follows_the_path_of_the/
+
+import shapefile as shp
+import pandas as pd
+
+from bokeh.plotting import figure, show
+from bokeh.palettes import YlOrRd5
+from bokeh.models import ColumnDataSource, LinearColorMapper, Label, ColorBar
+from bokeh.sampledata.us_states import data
+
+states = pd.DataFrame.from_dict(data, orient="index")
+states.drop(["AK", "HI"], inplace=True)
+
+trends = pd.read_csv("eclipse_data/trends.csv")
+
+states.set_index("name", inplace=True)
+trends.set_index("Region", inplace=True)
+
+states["trend"] = trends["solar eclipse"]
+
+upath17 = shp.Reader("eclipse_data/upath17")
+(totality_path,) = upath17.shapes()
+
+p = figure(plot_width=1000, plot_height=600)
+
+p.grid.grid_line_color = None
+
+p.title.text = "Google Search Trends and the Path of Solar Eclipse, 21 August 2017"
+p.title.align = "center"
+p.title.text_font_size = "16pt"
+
+mapper = LinearColorMapper(palette=list(reversed(YlOrRd5)), low=0, high=100)
+
+source = ColumnDataSource(data=dict(
+    state_xs=list(states.lons),
+    state_ys=list(states.lats),
+    trend=states.trend,
+))
+us = p.patches("state_xs", "state_ys",
+    fill_color=dict(field="trend", transform=mapper),
```

Comment:
Is it possible to configure a transform without need for using explicit data source and field references?

Reply:
there is some initial work in the new `bokeh.transform` that we can expand upon